javascript/react/good-to-know/keyed-fragments-in-react.md

```diff
@@ -24,67 +24,133 @@ aspects:
 
 ---
 
-# Keyed fragments in **React**
+# Keyed fragments in React
 
 ---
 ## Content
 
-With two sets of children that need reordering, there is no way of putting a key on each set without adding a wrapper element.
-
-To solve this problem, **React** provides the `createFragment` add-on which can give keys to sets of children.
-
-```javascript
-Array<ReactNode> createFragment(
-    object children);
+Take a look at the following code:
+
+```jsx
+class Columns extends React.Component {
+  render() {
+    return (
+      <div>
+        <td>Is this real life</td>
+        <td>Or is it fantasy</td>
+      </div>
+    );
+  }
+}
+
+class Table extends React.Component {
+  render() {
+    return (
+      <table>
+        <tr>
+          <Columns />
+        </tr>
+      </table>
+    );
+  }
+}
 ```
 
-Instead of:
-```javascript
-children = [this.props.rightChildren,
-    this.props.leftChildren];
-```
-This can be done like:
-```javascript
-var createFragment= require(
-    'react-addons-create-fragment');
-
-children = createFragment({
-  right: this.props.rightChildren,
-  left:this.props.leftChildren,
-});
+Although it might seem correct at a first glance, this would result in an invalid HTML code because the `<td>`s in the `Columns` component are encapsulated in a `<div>`. The solution to this problem is represented by `Fragment`s, which allows you to add a parent tag to your JSV elements without adding an extra node to the DOM.
```

Comment:
```suggestion
Although it might seem correct at a first glance, this would result in an invalid HTML code because the `<td>`s in the `Columns` component are encapsulated in a `<div>`. One solution to this problem is represented by `Fragment`s, which allows you to add a parent tag to your JSX elements without adding an extra node to the DOM.
```